sdks/python/apache_beam/typehints/trivial_inference_test.py

```diff
@@ -235,6 +235,60 @@ def m(self, x):
     self.assertReturnType(typehints.Any, lambda: A.m(A(), 3.0), depth=0)
     self.assertReturnType(float, lambda: A.m(A(), 3.0), depth=1)
 
+  # pylint: disable=eval-used
+  # TODO(udim): Remove eval() call once Python 2 support is removed from Beam.
+  @unittest.skipIf(sys.version_info < (3,), 'Python 3 only')
+  def testBuildListUnpack(self):
+    # Lambda uses BUILD_LIST_UNPACK opcode in Python 3.
+    # Uses eval() to hide Python 3 syntax from Python 2.
+    eval('''self.assertReturnType(typehints.List[int],
+                                  lambda _list: [*_list, *_list, *_list],
+                                  [typehints.List[int]])''')
+
+  # TODO(udim): Remove eval() call once Python 2 support is removed from Beam.
+  @unittest.skipIf(sys.version_info < (3,), 'Python 3 only')
+  def testBuildTupleUnpack(self):
+    # Lambda uses BUILD_TUPLE_UNPACK opcode in Python 3.
+    # Uses eval() to hide Python 3 syntax from Python 2.
+    eval('''self.assertReturnType(
+                typehints.Tuple[int, str, str],
+                lambda _list1, _list2: (*_list1, *_list2, *_list2),
+                [typehints.List[int], typehints.List[str]])''')
+  # pylint: enable=eval-used
+
+  def testBuildTupleUnpackWithCall(self):
+    # Lambda uses BUILD_TUPLE_UNPACK_WITH_CALL opcode in Python 3.6, 3.7.
+    def fn(x1, x2, *unused_args):
+      return x1, x2
+
+    self.assertReturnType(typehints.Tuple[str, float],
+                          lambda x1, x2, _list: fn(x1, x2, *_list),
+                          [str, float, typehints.List[int]])
+    # No *args
+    self.assertReturnType(typehints.Tuple[str, typehints.List[int]],
+                          lambda x1, x2, _list: fn(x1, x2, *_list),
+                          [str, typehints.List[int]])
+
+  @unittest.skipIf(sys.version_info < (3, 6), 'CALL_FUNCTION_EX is new in 3.6')
+  def testCallFunctionEx(self):
+    # Test when fn arguments are built using BUiLD_LIST.
```

Comment:
nit: BUILD_LIST

Reply:
I'll leave this for my next PR as I don't want to reset the tests.